{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Imports & File Paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import subprocess\n",
    "\n",
    "RAW_DATA_DIR = \"../data/raw/\"\n",
    "PROCESSED_DATA_DIR = \"../data/processed/EUR/\"\n",
    "PLINK_PATH = \"../plink\"\n",
    "\n",
    "sample_file = f\"{RAW_DATA_DIR}synthetic_v1.sample\"\n",
    "fam_file = f\"{RAW_DATA_DIR}synthetic_v1.fam\"\n",
    "pheno_file = f\"{RAW_DATA_DIR}synthetic_v1.pheno3\"\n",
    "chr_plink_path = f\"{RAW_DATA_DIR}synthetic_v1_chr-\"\n",
    "\n",
    "filtered_fam = f\"{PROCESSED_DATA_DIR}synthetic_v1_filtered.fam\"\n",
    "filtered_pheno = f\"{PROCESSED_DATA_DIR}synthetic_v1_filtered.pheno\"\n",
    "filtered_chr_plink_path = f\"{PROCESSED_DATA_DIR}synthetic_v1_filtered_chr-\"\n",
    "\n",
    "chromosomes = list(range(1, 23))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2 Preprocessing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.2 Load `.sample` and `.fam` files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(  Ancestry   IID\n",
       " 0      AFR  syn1\n",
       " 1      AFR  syn2\n",
       " 2      AFR  syn3\n",
       " 3      AFR  syn4\n",
       " 4      AFR  syn5,\n",
       "     FID   IID  Father  Mother  Sex  Phenotype\n",
       " 0  syn1  syn1       0       0    0         -9\n",
       " 1  syn2  syn2       0       0    0         -9\n",
       " 2  syn3  syn3       0       0    0         -9\n",
       " 3  syn4  syn4       0       0    0         -9\n",
       " 4  syn5  syn5       0       0    0         -9)"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_df = pd.read_csv(sample_file, sep='\\s+', header=None, names=[\"Ancestry\"])\n",
    "\n",
    "fam_cols = [\"FID\", \"IID\", \"Father\", \"Mother\", \"Sex\", \"Phenotype\"]\n",
    "fam_df = pd.read_csv(fam_file, sep='\\s+', names=fam_cols)\n",
    "\n",
    "sample_df[\"IID\"] = fam_df[\"IID\"].values\n",
    "\n",
    "sample_df.head(), fam_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.3 Filter individuals by ancestry"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>Father</th>\n",
       "      <th>Mother</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Phenotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>504000</th>\n",
       "      <td>syn504001</td>\n",
       "      <td>syn504001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504001</th>\n",
       "      <td>syn504002</td>\n",
       "      <td>syn504002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504002</th>\n",
       "      <td>syn504003</td>\n",
       "      <td>syn504003</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504003</th>\n",
       "      <td>syn504004</td>\n",
       "      <td>syn504004</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504004</th>\n",
       "      <td>syn504005</td>\n",
       "      <td>syn504005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              FID        IID  Father  Mother  Sex  Phenotype\n",
       "504000  syn504001  syn504001       0       0    0         -9\n",
       "504001  syn504002  syn504002       0       0    0         -9\n",
       "504002  syn504003  syn504003       0       0    0         -9\n",
       "504003  syn504004  syn504004       0       0    0         -9\n",
       "504004  syn504005  syn504005       0       0    0         -9"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "selected_group = \"EUR\"\n",
    "filtered_ids = sample_df[sample_df['Ancestry'] == selected_group][\"IID\"].tolist()\n",
    "filtered_fam_df = fam_df[fam_df[\"IID\"].isin(filtered_ids)]\n",
    "\n",
    "filtered_fam_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.4 Load and merge `.pheno3` (phenotype file). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/qw/l_8mqcwx11s020m7gp2bhhzr0000gn/T/ipykernel_38783/3284933130.py:1: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "  pheno_df = pd.read_csv(pheno_file, delim_whitespace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Phenotype\n",
      "2    84065\n",
      "1    83935\n",
      "Name: count, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>Father</th>\n",
       "      <th>Mother</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Phenotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>syn504001</td>\n",
       "      <td>syn504001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>syn504002</td>\n",
       "      <td>syn504002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>syn504003</td>\n",
       "      <td>syn504003</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>syn504004</td>\n",
       "      <td>syn504004</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>syn504005</td>\n",
       "      <td>syn504005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         FID        IID  Father  Mother  Sex  Phenotype\n",
       "0  syn504001  syn504001       0       0    0          2\n",
       "1  syn504002  syn504002       0       0    0          2\n",
       "2  syn504003  syn504003       0       0    0          1\n",
       "3  syn504004  syn504004       0       0    0          2\n",
       "4  syn504005  syn504005       0       0    0          2"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pheno_df = pd.read_csv(pheno_file, delim_whitespace=True)\n",
    "\n",
    "pheno_df.rename(columns={\"Sample\": \"IID\"}, inplace=True)\n",
    "filtered_fam_df = fam_df[fam_df[\"IID\"].isin(filtered_ids)].merge(pheno_df[[\"IID\", \"Phenotype(binary)\"]], on=\"IID\", how=\"inner\")\n",
    "filtered_fam_df[\"Phenotype\"] = filtered_fam_df[\"Phenotype(binary)\"].map({0: 1, 1: 2})  # 1 = control, 2 = case\n",
    "filtered_fam_df = filtered_fam_df.drop(\"Phenotype(binary)\", axis=1)\n",
    "\n",
    "filtered_fam_df[[\"FID\", \"IID\", \"Phenotype\"]].to_csv(filtered_pheno, sep=\" \", index=False, header=False)\n",
    "filtered_fam_df.to_csv(filtered_fam, sep=\" \", index=False, header=False)\n",
    "\n",
    "print(filtered_fam_df[\"Phenotype\"].value_counts())\n",
    "filtered_fam_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.6 Filter based on filtered `.fam` file & QC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==== Filtering 1... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-1.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-1\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-1\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "533532 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-1.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "63210 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "470322 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-1.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-1.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-1.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 2... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-2.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-2\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-2\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "560980 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-2.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 7 variants removed due to Hardy-Weinberg exact test.\n",
      "62841 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "498132 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-2.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-2.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 3... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-3.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-3\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-3\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "474347 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-3.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "54809 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "419538 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-3.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-3.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-3.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 4... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-4.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-4\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-4\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "494539 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-4.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 3 variants removed due to Hardy-Weinberg exact test.\n",
      "55700 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "438836 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-4.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-4.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-4.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 5... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-5.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-5\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-5\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "417894 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-5.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is in [0.9999995, 1).\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 7 variants removed due to Hardy-Weinberg exact test.\n",
      "47221 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "370666 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-5.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-5.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-5.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 6... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-6.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-6\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-6\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "467670 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-6.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is in [0.9999995, 1).\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 4 variants removed due to Hardy-Weinberg exact test.\n",
      "52967 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "414699 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-6.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-6.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-6.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 7... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-7.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-7\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-7\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "399944 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-7.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 4 variants removed due to Hardy-Weinberg exact test.\n",
      "49788 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "350152 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-7.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-7.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-7.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 8... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-8.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-8\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-8\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "362505 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-8.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is in [0.9999995, 1).\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 3 variants removed due to Hardy-Weinberg exact test.\n",
      "39945 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "322557 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-8.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-8.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-8.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 9... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-9.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-9\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-9\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "309581 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-9.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is in [0.9999995, 1).\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 10 variants removed due to Hardy-Weinberg exact test.\n",
      "37119 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "272452 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-9.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-9.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-9.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 10... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-10.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-10\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-10\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "361561 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-10.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 3 variants removed due to Hardy-Weinberg exact test.\n",
      "43441 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "318117 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-10.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-10.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-10.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 11... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-11.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-11\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-11\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "333788 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-11.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 1 variant removed due to Hardy-Weinberg exact test.\n",
      "33207 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "300580 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-11.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-11.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-11.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 12... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-12.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-12\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-12\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "324050 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-12.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "34914 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "289136 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-12.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-12.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-12.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 13... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-13.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-13\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-13\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "254278 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-13.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is in [0.9999995, 1).\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "25677 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "228601 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-13.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-13.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-13.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 14... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-14.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-14\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-14\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "229536 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-14.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is in [0.9999995, 1).\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 3 variants removed due to Hardy-Weinberg exact test.\n",
      "29058 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "200475 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-14.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-14.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-14.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 15... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-15.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-15\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-15\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "205489 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-15.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "25854 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "179635 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-15.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-15.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-15.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 16... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-16.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-16\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-16\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "211887 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-16.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "25597 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "186290 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-16.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-16.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-16.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 17... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-17.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-17\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-17\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "191820 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-17.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 1 variant removed due to Hardy-Weinberg exact test.\n",
      "25490 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "166329 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-17.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-17.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-17.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 18... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-18.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-18\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-18\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "197394 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-18.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "20355 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "177039 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-18.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-18.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-18.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 19... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-19.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-19\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-19\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "163414 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-19.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 1 variant removed due to Hardy-Weinberg exact test.\n",
      "19908 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "143505 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-19.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-19.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-19.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 20... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-20.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-20\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-20\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "153988 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-20.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "18854 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "135134 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-20.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-20.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-20.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 21... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-21.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-21\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-21\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "109673 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-21.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "14494 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "95179 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-21.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-21.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-21.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "==== Filtering 22... ====\n",
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../data/processed/EUR/synthetic_v1_filtered_chr-22.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../data/raw/synthetic_v1_chr-22\n",
      "  --geno 0.05\n",
      "  --hwe 1e-6\n",
      "  --keep ../data/processed/EUR/synthetic_v1_filtered.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out ../data/processed/EUR/synthetic_v1_filtered_chr-22\n",
      "  --pheno ../data/processed/EUR/synthetic_v1_filtered.pheno\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "116524 variants loaded from .bim file.\n",
      "1008000 people (0 males, 0 females, 1008000 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-22.nosex .\n",
      "168000 phenotype values present after --pheno.\n",
      "--keep: 168000 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 168000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "16457 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "100067 variants and 168000 people pass filters and QC.\n",
      "Among remaining phenotypes, 84065 are cases and 83935 are controls.\n",
      "--make-bed to ../data/processed/EUR/synthetic_v1_filtered_chr-22.bed +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-22.bim +\n",
      "../data/processed/EUR/synthetic_v1_filtered_chr-22.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "for chrom in chromosomes:\n",
    "    print(f\"==== Filtering {chrom}... ====\")\n",
    "    plink_filter_cmd = [\n",
    "        PLINK_PATH, '--bfile', f'{chr_plink_path}{chrom}',\n",
    "        '--keep', filtered_fam,\n",
    "        '--pheno', filtered_pheno,\n",
    "        '--geno', '0.05',\n",
    "        '--hwe', '1e-6',\n",
    "        '--maf', '0.01',\n",
    "        '--allow-no-sex',\n",
    "        '--make-bed',\n",
    "        '--out', f'{filtered_chr_plink_path}{chrom}'\n",
    "    ]\n",
    "    subprocess.run(plink_filter_cmd)\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
